src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,540 @@
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.1
+import "../../core"
+import GCompris 1.0
+import "dataset.js" as Dataset
+
+ActivityBase {
+    id: activity
+    property string url : "qrc:/gcompris/src/activities/morse_code/resource/"
+    onStart: focus = true
+    onStop: {}
+
+
+    pageComponent: Image {
+        id: background
+        source: url+"background.svg"
+        fillMode: Image.PreserveAspectCrop
+        sourceSize.width: Math.max(parent.width, parent.height)
+
+        signal start
+        signal stop
+
+        Component.onCompleted: {
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property alias background: background
+            property alias bar: bar
+            property alias bonus: bonus
+            property alias score: score
+            property alias textInput: textInput
+            property var dataset: Dataset.get()
+            property bool toAlpha: dataset[currentLevel].toAlpha
+            property string questionText: dataset[currentLevel].question
+            property string questionValue
+            property string instruction: dataset[currentLevel].instruction
+            property int currentLevel: 0
+            property int numberOfLevel: dataset.length
+
+            onToAlphaChanged: {
+                textInput.text = ''
+                morseConverter.alpha=0
+                if(toAlpha)
+                    keyboard.populateAlpha()
+                else
+                    keyboard.populateMorse()
+            }
+
+            function start() {
+                if (!ApplicationInfo.isMobile)
+                    textInput.forceActiveFocus();
+                items.currentLevel = 0
+                initLevel()
+            }
+
+            function initLevel() {
+                score.currentSubLevel = 1
+                score.numberOfSubLevels = dataset[currentLevel].values[1].length
+                initSubLevel()
+            }
+
+            function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    questionValue = questionValue.replace(/\./g, '\u00B7');
```

Comment:
instead of having this '\u00B7' everywhere, it would be best to do a readonly variable